MVC Badge System/MVC Badge System/Views/Gift/GridIndex.cshtml

```diff
@@ -0,0 +1,102 @@
+﻿@model IEnumerable<MVC_Badge_System.Models.Gift>
+
+@{
+    ViewBag.Title = "GridIndex";
+}
+
+<h2>GridIndex</h2>
+
+<p>
+    @Html.ActionLink("Create New", "Create")
+</p>
+<table class="table">
+    <tr>
+        @for (int i = 0; i <= 6; i++)
+        {
+            <td>
+                <button type="button" height="80%" width="80%"><img src="http://rs519.pbsrc.com/albums/u359/hoobity/1324749076815.gif~c200" /></button>
+            </td>
+        }
+    </tr>
+
+    @for (int i = 0; i <= 30;)
+    {
+        <tr>
+            @for (int j = 0; j <= 6; j++)
+            {
+                if (i <= 30)
+                {
+                <td>
+                    <button type="button" height="90%" width="90%"><img src="http://rs519.pbsrc.com/albums/u359/hoobity/1324749076815.gif~c200" /></button>
+                </td>
+                } else
+                {
+                    <td></td>
+                }
+                i++;
+            }
+        </tr>
+    }
+
+    @*actual table*@
+    @*
+    <tr>
+        @foreach (var coreBadge in Model) {
+            if (Model.Badge.Type == "core")
```

Comment:
Does it make sense to do `@foreach (var coreBadge in Model) {` and then validate `if (Model.Badge.Type)`?